File: UNXNavigatorDemo/Feature/Basic/Controllers/ViewController02_BackgroundImage.m
//
//  ViewController02_BackgroundImage.m
//  UNXNavigatorDemo
//
//  Created by Leo Lee on 2020/10/25.
//

#import <UNXNavigator/UNXNavigator.h>

#import "ViewController02_BackgroundImage.h"
#import "UIImage+NavigationBar.h"

@interface ViewController02_BackgroundImage ()

@end

@implementation ViewController02_BackgroundImage

- (void)viewDidLoad {
    [super viewDidLoad];
    
}

- (UIStatusBarStyle)preferredStatusBarStyle {
    return UIStatusBarStyleLightContent;
}

- (UIColor *)unx_barTintColor {
    return [UIColor whiteColor];
}

- (NSDictionary<NSAttributedStringKey,id> *)unx_titleTextAttributes {
    return @{NSForegroundColorAttributeName: [self unx_barTintColor]};
}

- (UIImage *)unx_navigationBarBackgroundImage {
    return UIImage.navigationBarBackgorundImage;
}

@end

File: Example/UNXNavigatorSwiftDemo/Basic/ViewController02_BackgroundImage.swift
//
//  ViewController02_BackgroundImage.swift
//  UNXNavigatorSwiftDemo
//
//  Created by Leo Lee on 2020/11/23.
//

import UIKit

class ViewController02_BackgroundImage: BaseTableViewController {

    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }
    
    override var preferredStatusBarStyle: UIStatusBarStyle {
        return .lightContent
    }
    
    override var unx_barTintColor: UIColor? {
        return .white
    }
    
    override var unx_titleTextAttributes: [NSAttributedString.Key : Any]? {
        return [NSAttributedString.Key.foregroundColor: UIColor.white]
    }
    
    override var unx_navigationBarBackgroundImage: UIImage? {
        return UIImage.navigationBarBackgorund
    }

}
